src/nos/adapters/inbound/event_sub.py

```diff
@@ -83,35 +83,22 @@ def __init__(
         ]
         self._config = config
         self._orchestrator = orchestrator
-
-    async def _notify_access_request_created(self, payload: JsonObject) -> None:
-        """Notify that an access request was created"""
-        validated_payload = get_validated_payload(payload, AccessRequestDetails)
-        await self._orchestrator.process_access_request_created(
-            user_id=validated_payload.user_id, dataset_id=validated_payload.dataset_id
-        )
-
-    async def _notify_access_request_allowed(self, payload: JsonObject) -> None:
-        """Notify that an access request was allowed"""
-        validated_payload = get_validated_payload(payload, AccessRequestDetails)
-        await self._orchestrator.process_access_request_allowed(
-            user_id=validated_payload.user_id, dataset_id=validated_payload.dataset_id
-        )
-
-    async def _notify_access_request_denied(self, payload: JsonObject) -> None:
-        """Notify that an access request was denied"""
+        self._access_request_map = {
+            config.access_request_created_type: self._orchestrator.process_access_request_created,
+            config.access_request_allowed_type: self._orchestrator.process_access_request_allowed,
+            config.access_request_denied_type: self._orchestrator.process_access_request_denied,
+        }
+
```

Comment:
I like the thinking, but those kinds of conventions usually make for a few miserable hours of debugging later on. In [1ba97d2](https://github.com/ghga-de/notification-orchestration-service/pull/1/commits/1ba97d26dd397dcfbf697b5fa219ea43afdf7a08), I funnel events for explicit types into one gatekeeper function, which dispatches the payload accordingly from within the core. Simplifies both modules and won't give me a headache six months from now.

Reply:
Whatever gives us the least headache is good. The idea here is to avoid duplication and the need to change things in several places if you add a new kind of event. But this still can be refactored when we add more events and find that things will become cumbersome or error prone.